### ISSUE TITLE
Scrubbing mission doesn't unassign hardware

### ISSUE THREAD

**Issue by rnyoakum** [raceintospace/raceintospace#292]
TLDR: Scrubbing a mission does not unassign mission hardware and make it available for use on other missions.

Full explanation:
I had a pair of missions, a Manned Suborbital and an Orbital Satellite, planned for my turn. I checked the hardware requirements in the VAB: an Atlas + Mercury for the manned mission and an Atlas + Explorer for the Orbital Satellite. I had none of the hardware on hand. I autopurchased the Atlas + Explorer and assigned it to the Orbital Satellite.

Next, I visited the Launch Facility B and scrubbed the Orbital Satellite. Returning to the VAB, I tried to assign hardware to the Manned Suborbital. As I suspected, the Atlas rocket was not available. I checked the rocket count in Hardware Purchasing, which confirmed I did indeed have 1 Atlas rocket. Scrubbing the Orbital Satellite did not free up the assigned hardware, contrary to the popup alert.

**Comment by peyre** [comment]
Yeah, that's not good.  In addition, the game subtracts hardware for missions that fail to fly.  If you have a launch that's canceled because of a last-minute issue with the boosters, for instance, the rocket, capsule, etc. is all subtracted.  Or if you have a Joint mission, part A launches but fails so part B doesn't fly, you still lose the hardware for part B.

**Comment by rnyoakum** [comment]
Resolved by commit bce816004721a4b772b9b3ff53ebf36903172593.

**Comment by peyre** [comment]
Great work!  Does this have any effect on missions that don't fly (like if they have a last-second abort on the boosters) not freeing up the hardware?

**Comment by rnyoakum** [comment]
No, that gets handled separately. Scrubbing the mission removes the 'hold' the mission placed on the hardware in the VAB. The VAB tracks how much of your equipment inventory is "spoken for", but doesn't change how many units you have. Launching a mission actually depletes your supply and in the game the 'abort' is a specific fail-state of the mission. So while the two situations are logically the same, they're treated entirely differently.

My fix should cover scrubbing the mission up to and including the last chance to scrub the mission right before launch. Everything after you push "Continue" to start the launch countdown is an actual mission launch, and treated as an attempt. Consequently, issue #149 will require implementing special case checks in the mission code, or a more clever rewriting of how missions are handled. At this point, I haven't looked at the mission execution code enough to know the best approach.

**Comment by peyre** [comment]
Yes, that makes sense, thanks.  Hopefully we'll get that fixed as well, eventually.
